src/main/java/dev/symphony/harmony/mixin/transportation/HorseArmorPreventsBucking.java

```diff
@@ -0,0 +1,47 @@
+package dev.symphony.harmony.mixin.transportation;
+
+import dev.symphony.harmony.config.HarmonyConfig;
+import it.unimi.dsi.fastutil.objects.Object2FloatArrayMap;
+import net.minecraft.entity.passive.AbstractHorseEntity;
+import net.minecraft.inventory.Inventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Map;
+
+@Mixin (AbstractHorseEntity.class)
+public class HorseArmorPreventsBucking {
+
+    // FEATURE:
+    // AUTHORS: KikuGie, Flatkat
+    @Unique
+    private static final Map<Item, Float> preventBuckingChance = new Object2FloatArrayMap<>();
```

Comment:
Pretty sure there are better ways to instantiate a map than `Object2FloatArrayMap`, mostly, because things such as "additional methods that use polymorphism" aren't needed here, but it does work, so it's not worth it to replace, just keep the lighter alternatives in mind for the future, not much performance difference, but it's just a better practice to use simpler classes.